cyraft/node.py

```diff
@@ -380,25 +397,8 @@ def _append_entries_processing(
         This method is called when the append entries request passes all checks and can be appended to the log.
         """
         assert len(request.log_entry) == 1  # in our implementation only a single entry is sent at a time
-
-        # If an existing entry conflicts with a new one (same index but different terms),
-        # delete the existing entry and all that follow it (§5.3)
         new_index = request.prev_log_index + 1
         _logger.info(c["append_entries"] + "Node ID: %d -- new_index: %d" + c["end_color"], self._node.id, new_index)
-        for log_index, log_entry in enumerate(self._log[1:]):
-            if (
-                log_index + 1  # index + 1 because we skip the first entry (self._log[1:])
-            ) == new_index and log_entry.term != request.log_entry[0].term:
-                _logger.info(
-                    c["append_entries"] + "Node ID: %d -- deleting from: %d" + c["end_color"],
-                    self._node.id,
-                    log_index + 1,
-                )
-                number_of_entries_to_delete = len(self._log) - (log_index + 1)
-                self._next_index = [x - number_of_entries_to_delete for x in self._next_index]
-                del self._log[log_index + 1 :]
-                self._commit_index = log_index
-                break
```

Comment:
`_append_entries_processing` does not require this change, as discussed [here](https://github.com/OpenCyphal-Garage/cyraft/pull/11#discussion_r1682389382).